### PR TITLE
Update common workflow

### PR DESCRIPTION
**Pull request by cicdguy**
Because we're now using a single-branch GitHub Flow strategy, we don't need other branches that we need to act on.

We also don't want to build pages for `main`.

### CODE REVIEW

**Comment by ddsjoberg** [comment]
@cicdguy we may still want to build the pkgdown site on main. There is a pkgdown option to build the dev site and save it in a subfolder named `dev`. I think this newly option is turned on in one of the open PRs.

Is the primary site (latest release) now being hosted from the root of the `gh-pages` branch?

File: .github/workflows/common.yml [hunk 1]
```diff
@@ -15,17 +15,11 @@ on:
       - "v*"
     branches:
       - main
-      - devel
-      - patch
```
Comment (bms63) [inline .github/workflows/common.yml:19]: we actually need the patch one as we still don't want to release everything from `main` when doing a patch/bug fix

Comment (ddsjoberg) [reply .github/workflows/common.yml:19]: After you release from patch, you'd merge to main and the site would be built then. So do we need the patch pkgdown builds?

Comment (bms63) [reply .github/workflows/common.yml:19]: but i think we still want the workflows to trigger when PRing into `patch`?  IF we remove `patch` then it won't trigger any of the checks - I thought??

**Review by bms63 — APPROVED** [review]
Let's see what happens!

**Comment by bundfussr** [comment]
@cicdguy , @bms63 , do we have a separate PR to run the checks in `patch` (and for releases) with the latest released version of the upstream dependencies (instead of the version from `main`)? Or is this already solved?

**Comment by bms63** [comment]
The checks are running when we PR into `patch`.  This update/PR would of removed them, but patch was added back in.

We no longer make the `main` version of the website - which might impact the `dev` version being created.  Not completely sure!!

**Pull request merged**